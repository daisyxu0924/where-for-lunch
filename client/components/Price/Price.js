import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Price.css';

export default class Price extends PureComponent {
  static propTypes = {
    price: PropTypes.string,
    onChange: PropTypes.func,
  };

  handleOnChangeAction = (e) => {
    const { price } = this.props;
    const value = price.length ? price.split(',') : [];
    const index = value.indexOf(e);
    if (index > -1) value.splice(index, 1);
    else value.push(e);
    this.props.onChange(value.toString());
  }

  render() {
    const { price } = this.props;
    const value = price.split(',');
    const options = ['$', '$$', '$$$', '$$$$'];
    return (
      <div className={styles.root}>
        <div>price range:</div>
        <div>
          {options.map((item, index) => (
            <span key={item}>
              <input type="checkbox" onChange={() => this.handleOnChangeAction((index + 1).toString())} checked={value.includes((index + 1).toString())}></input>
              <span>{item}</span>
            </span>
          ))}
        </div>
      </div>
    );
  }
}
