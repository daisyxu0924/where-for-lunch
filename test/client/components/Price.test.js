import React from 'react';
import { shallow } from 'enzyme';
import Price from 'client/components/Price/Price';
import renderer from 'react-test-renderer';

test('With Enzyme, Price component will show 3 checks', () => {
  const price = '3,4';
  const wrapper = shallow(
    <Price price={price} />,
  );
  let total = 0;
  const b = wrapper.find({ type: 'checkbox' });
  b.map((item) => {
    if (item.props().checked) total += 1;
    return false;
  });
  expect(total).toBe(2);
});

test('With Jest snapshot, Button component renders with disable property', () => {
  const price = '3,4';
  const b = renderer
    .create(<Price price={price} />)
    .toJSON();
  expect(b).toMatchSnapshot();
});
