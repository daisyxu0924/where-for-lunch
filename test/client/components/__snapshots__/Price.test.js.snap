// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, Button component renders with disable property 1`] = `
<div
  className="root"
>
  <div>
    price range:
  </div>
  <div>
    <span>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        $
      </span>
    </span>
    <span>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        $$
      </span>
    </span>
    <span>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        $$$
      </span>
    </span>
    <span>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span>
        $$$$
      </span>
    </span>
  </div>
</div>
`;
